application/common/sync/Synchronizer.php

```diff
@@ -112,8 +121,55 @@ public function syncAll()
         }
         
         // Deactivate the remaining (unprocessed) users in the ID Broker list.
-        foreach (array_keys($idBrokerUserInfoByEmployeeId) as $employeeId) {
-            $this->deactivateUser($employeeId);
+        foreach ($idBrokerUserInfoByEmployeeId as $employeeId => $userInfo) {
+            // If this user not currently inactive, deactivate them.
+            if ($userInfo['active'] !== 'no') {
+                $employeeIdsToDeactivate[] = $employeeId;
+            }
+        }
+        
+        /** @todo Add a safety check here to avoid deactivating too many users. */
+        
+        foreach ($usersToAdd as $userToAdd) {
+            try {
+                $this->idBroker->createUser($userToAdd->toArray());
+            } catch (Exception $e) {
+                Yii::error(sprintf(
+                    'Failed to add user to ID Broker (Employee ID: %s). '
+                    . 'Error %s: %s',
+                    var_export($userToAdd->employeeId, true),
+                    $e->getCode(),
+                    $e->getMessage()
+                ));
+            }
+        }
+        
+        foreach ($usersToUpdateAndActivate as $userToUpdateAndActivate) {
+            try {
+                $this->activateAndUpdateUser($userToUpdateAndActivate);
+            } catch (Exception $e) {
+                Yii::error(sprintf(
+                    'Failed to update/activate user in the ID Broker (Employee ID: %s). '
+                    . 'Error %s: %s',
+                    var_export($userToUpdateAndActivate->employeeId, true),
+                    $e->getCode(),
+                    $e->getMessage()
+                ));
+            }
+        }
+        
+        foreach ($employeeIdsToDeactivate as $employeeIdToDeactivate) {
```

Comment:
based on L147, I would've expected `$usersToDeactivate as $userToDeactivate`.  Was the deviation deliberate?